Adds  quickstarts for Orchestrate. (#1978)
saucelabs/sauce-docs: docs/orchestrate/quickstart-webdriverio.md

---

> id: webdriverio-orchestrate-quickstart

Reviewer: ```suggestion
id: quickstart-webdriverio
```

title: WebdriverIO Quickstart
sidebar_label: WebdriverIO Quickstart
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import useBaseUrl from '@docusaurus/useBaseUrl';

This page explains how to run your WebdriverIO tests in Sauce Orchestrate with `saucectl`.

## What You'll Need

- A Sauce Labs account (if you don't have one, start a [free trial](https://saucelabs.com/sign-up))
- Your Sauce Labs [Username and Access Key](https://app.saucelabs.com/user-settings)
- [Node.js](https://nodejs.org/en/) to use the NPM package manager
- A [GitHub](https://github.com/signup?ref_cta=Sign+up&ref_loc=header+logged+out&ref_page=%2F&source=header-home) account
- [Git](https://git-scm.com/downloads)
- Access to a Docker registry. This example uses [DockerHub](https://hub.docker.com).
- [Docker installed](https://docs.docker.com/engine/install/) if you want to modify and rebuild the example docker image.

## How Orchestrate Works

> Sauce Orchestrate uses container technology to package and run your tests in the Sauce Labs cloud. See the [Getting Started > How it Works section](/docs/orchestrate/getting-started/#how-it-works) for more details.

Author: ```suggestion
Sauce Orchestrate uses container technology to package and run your tests in the Sauce Labs cloud. See the [Getting Started > How it Works section](/docs/orchestrate/getting-started.md#how-it-works) for more details.
```

> Sauce Orchestrate uses container technology to package and run your tests in the Sauce Labs cloud. See the [Getting Started > How it Works section](/docs/orchestrate/getting-started.md#how-it-works) for more details.

Reviewer: ```suggestion
Sauce Orchestrate uses container technology to package and run your tests in the Sauce Labs cloud. See [How it Works](/docs/orchestrate/getting-started/#how-it-works) for more details.
```

> Sauce Orchestrate uses container technology to package and run your tests in the Sauce Labs cloud. See the [Getting Started > How it Works section](/docs/orchestrate/getting-started.md#how-it-works) for more details.

Reviewer: ```suggestion
Sauce Orchestrate uses container technology to package and run your tests in the Sauce Labs cloud. See [How it Works](/docs/orchestrate/getting-started/#how-it-works) for more details.
```

> The following example uses a project with a `saucectl` configuration file that is set up to run in Orchestrate. See [`saucectl` Configuration](/docs/orchestrate/saucectl-configuration.md) for more information about the configuration and run options available. The example also has a Dockerfile for building the container image. See [Building Images](/docs/orchestrate/building-images.md) for more information about creating your own container images.

Reviewer: ```suggestion
The following example uses a project with a `saucectl` configuration file that is set up to run in Orchestrate. See [`saucectl` Configuration](/docs/orchestrate/saucectl-configuration) for more information about the configuration and run options available. The example also has a Dockerfile for building the container image. See [Building Images](/docs/orchestrate/building-images) for more information about creating your own container images.
```

## Step 1: Clone the Project

<Tabs
defaultValue="https"
values={[
{label: 'HTTPS', value: 'https'},
{label: 'SSH', value: 'ssh'},
]}>

<TabItem value="https">

```
git clone https://github.com/saucelabs/saucectl-imagerunner-example.git
```

</TabItem>
<TabItem value="ssh">

```
git clone git@github.com:saucelabs/saucectl-imagerunner-example.git
```

</TabItem>
</Tabs>

## Step 2: Link Your Sauce Labs Account

`saucectl` requires access to a valid Sauce Labs account.

:::tip Use Environment Variables

> `saucectl` detects your Sauce Labs credentials [environment variables](/basics/environment-variables) and prioritizes them over values in the `credentials.yml` file when both are present. **If you have set them, you may skip this step.** Not sure if you have them set? Run the following command to check:

Reviewer: ```suggestion
`saucectl` uses your Sauce Labs credentials as [environment variables](/basics/environment-variables) and prioritizes them over values in the `credentials.yml` file when both are present. **If you have set them, you may skip this step.** Not sure if you have them set? Run the following command to check:
```
